这个似乎要修改 C++ 侧?

```
2023-11-06 01:20:58 ======================================================================
2023-11-06 01:20:58 ERROR: test_check_grad (test_unfold_op.TestUnfoldOp)
2023-11-06 01:20:58 ----------------------------------------------------------------------
2023-11-06 01:20:58 Traceback (most recent call last):
2023-11-06 01:20:58   File "/workspace/Paddle/build/test/legacy_test/test_unfold_op.py", line 145, in test_check_grad
2023-11-06 01:20:58     self.check_grad(['X'], 'Y', check_pir=True)
2023-11-06 01:20:58   File "/workspace/Paddle/build/python/op_test.py", line 2880, in check_grad
2023-11-06 01:20:58     self.check_grad_with_place(
2023-11-06 01:20:58   File "/workspace/Paddle/build/python/op_test.py", line 3137, in check_grad_with_place
2023-11-06 01:20:58     new_ir_grad = self._get_ir_gradient(
2023-11-06 01:20:58   File "/workspace/Paddle/build/python/op_test.py", line 3677, in _get_ir_gradient
2023-11-06 01:20:58     outs = executor.run(
2023-11-06 01:20:58   File "/workspace/Paddle/build/python/paddle/base/executor.py", line 1637, in run
2023-11-06 01:20:58     res = self._run_pir_impl(
2023-11-06 01:20:58   File "/workspace/Paddle/build/python/paddle/base/executor.py", line 1952, in _run_pir_impl
2023-11-06 01:20:58     ret = new_exe.run(list(feed.keys()), return_numpy)
2023-11-06 01:20:58   File "/workspace/Paddle/build/python/paddle/base/executor.py", line 798, in run
2023-11-06 01:20:58     tensors = self._new_exe.run(feed_names)._move_to_list()
2023-11-06 01:20:58 ValueError: 
2023-11-06 01:20:58 
2023-11-06 01:20:58   Compile Traceback (most recent call last):
2023-11-06 01:20:58 
2023-11-06 01:20:58 --------------------------------------
2023-11-06 01:20:58 C++ Traceback (most recent call last):
2023-11-06 01:20:58 --------------------------------------
2023-11-06 01:20:58 0   paddle::framework::ThreadPoolTempl<paddle::framework::StlThreadEnvironment>::WorkerLoop(int)
2023-11-06 01:20:58 1   paddle::framework::PirInterpreter::RunInstructionBaseAsync(unsigned long)
2023-11-06 01:20:58 2   paddle::framework::PirInterpreter::RunInstructionBase(paddle::framework::InstructionBase*)
2023-11-06 01:20:58 3   paddle::framework::PhiKernelInstruction::Run()
2023-11-06 01:20:58 4   phi::KernelImpl<void (*)(phi::CPUContext const&, phi::DenseTensor const&, phi::DenseTensor const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, phi::DenseTensor*), &(void phi::UnfoldGradKernel<float, phi::CPUContext>(phi::CPUContext const&, phi::DenseTensor const&, phi::DenseTensor const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, phi::DenseTensor*))>::Compute(phi::KernelContext*)
2023-11-06 01:20:58 5   void phi::UnfoldGradKernel<float, phi::CPUContext>(phi::CPUContext const&, phi::DenseTensor const&, phi::DenseTensor const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, phi::DenseTensor*)
2023-11-06 01:20:58 6   phi::funcs::Col2ImFunctor<(phi::funcs::ColFormat)0, phi::CPUContext, float>::operator()(phi::CPUContext const&, phi::DenseTensor const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, phi::DenseTensor*, phi::DataLayout)
2023-11-06 01:20:58 7   float const* phi::DenseTensor::data<float>() const
2023-11-06 01:20:58 8   phi::enforce::EnforceNotMet::EnforceNotMet(phi::ErrorSummary const&, char const*, int)
2023-11-06 01:20:58 9   phi::enforce::GetCurrentTraceBackString[abi:cxx11](bool)
2023-11-06 01:20:58 
2023-11-06 01:20:58 ----------------------
2023-11-06 01:20:58 ----------------------
2023-11-06 01:20:58 Error Message Summary:
2023-11-06 01:20:58 ----------------------
2023-11-06 01:20:58 InvalidArgumentError: The type of data we are trying to retrieve (float32) does not match the type of data (float64) currently contained in the container.
2023-11-06 01:20:58   [Hint: Expected dtype() == phi::CppTypeToDataType<T>::Type(), but received dtype():11 != phi::CppTypeToDataType<T>::Type():10.] (at ../paddle/phi/core/dense_tensor.cc:164)
2023-11-06 01:20:58   [operator < pd_kernel.phi_kernel > error]
2023-11-06 01:20:58 
2023-11-06 01:20:58 ----------------------------------------------------------------------
```

--- a/test/legacy_test/test_unfold_op.py
+++ b/test/legacy_test/test_unfold_op.py
@@ -139,7 +139,7 @@ class TestUnfoldOp(OpTest):
         self.dtype = np.float64
 
     def test_check_output(self):
-        self.check_output()
+        self.check_output(check_pir=True)
 
     def test_check_grad(self):
         self.check_grad(['X'], 'Y', check_pir=True)

